src/vmm/src/vstate/vcpu/aarch64.rs

```diff
@@ -130,6 +133,15 @@ impl KvmVcpu {
         if self.index > 0 {
             kvi.features[0] |= 1 << kvm_bindings::KVM_ARM_VCPU_POWER_OFF;
         }
+
+        // If the host supports Pointer Authentication, enable it for the guest as well.
+        if KvmContext::check_capability(Cap::ArmPtrAuthAddress).map_err(Error::GetCapability)? {
+            kvi.features[0] |= 1 << kvm_bindings::KVM_ARM_VCPU_PTRAUTH_ADDRESS;
+        }
+        if KvmContext::check_capability(Cap::ArmPtrAuthGeneric).map_err(Error::GetCapability)? {
+            kvi.features[0] |= 1 << kvm_bindings::KVM_ARM_VCPU_PTRAUTH_GENERIC;
+        }
```

Comment:
nit: In KVM both `KVM_ARM_VCPU_PTRAUTH_ADDRESS` and `KVM_ARM_VCPU_PTRAUTH_GENERIC` are required in order for PAC to work, so you can convert this into
```rust
        if KvmContext::check_capability(Cap::ArmPtrAuthAddress).map_err(Error::GetCapability)? &&
           KvmContext::check_capability(Cap::ArmPtrAuthGeneric).map_err(Error::GetCapability)? {
                  kvi.features[0] |= 1 << kvm_bindings::KVM_ARM_VCPU_PTRAUTH_ADDRESS;
                  kvi.features[0] |= 1 << kvm_bindings::KVM_ARM_VCPU_PTRAUTH_GENERIC;
        }
```
It does not make much sense to enable them one by one.

Reply:
Well, we only care when they both present. But ultimately your version is fine as well.

Reply:
The doc: https://www.kernel.org/doc/html/latest/arm64/pointer-authentication.html